src/lib/slide-toggle/slide-toggle-config.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {InjectionToken} from '@angular/core';
+
+
+/**
+ * Slide toggle click action when user click on input element.
+ * noop: Do not toggle checked.
+ * check: Only toggle checked status. Default behavior
+ * undefined: Same as `check`.
+ */
+export type MatSlideToggleClickAction = 'noop' | 'check' | undefined;
+
+/**
+ * Injection token that can be used to specify the slide toggle click behavior.
+ */
+export const MAT_SLIDE_TOGGLE_CLICK_ACTION =
```

Comment:
Could you combine these two into one token `MAT_SLIDE_TOGGLE_DEFAULT_OPTIONS`? We use that pattern on a handful of other components so it would be a bit more consistent (see `MAT_AUTOCOMPLETE_DEFAULT_OPTIONS` for an example)